vscode/src/completions/inline-completion-item-provider.ts

```diff
@@ -695,3 +699,78 @@ function onlyCompletionWidgetSelectionChanged(prev: CompletionRequest, next: Com
 
     return prevSelectedCompletionInfo.text !== nextSelectedCompletionInfo.text
 }
+
+/**
+ * Handles showing an in-editor decoration when a first completion is accepted.
+ */
+class FirstCompletionDecorationHandler {
+    /**
+     * Duration to show decoration before automatically hiding.
+     *
+     * Modifying the document will also immediately hide.
+     */
+    private static readonly decorationDurationMilliseconds = 3000
```

Comment:
Given it auto-hides, a longer value is probably better.

```suggestion
    private static readonly decorationDurationMilliseconds = 10000
```

I gave 10s a go locally and it seemed good… unless there's some scenarios you have in mind where that would cause confusion?

Reply:
sgtm - I didn't think much about the number I just thought it made sense to hide after some time.